WordPress/gutenberg
# Changelog: append a generated list of first time contributors by PR label

## Description

This PR attempts to address the first few suggestions on [this discussion](https://github.com/WordPress/gutenberg/discussions/38244) regarding Establish a first-time contributor props culture in Gutenberg.

It's pretty basic and requires user intervention in terms of applying the correct labels in Github. 

The PR aims to:

- find all PRs merged with the label `First-time contributor`.
- find the user who created the PR.
- add them to a list of contributors under the title `First time contributors`.

This list is appended to the bottom of the existing Changelog and looks like this:

```
## First time contributors

The following PRs were merged by first time contrbutors:

- @alanjacobmathew: [Type] Documentation  Updating Faq.md typo. ([38154](https://github.com/WordPress/gutenberg/pull/38154))
- @imangm: RangeControl: Fix space between icons and rail. ([36935](https://github.com/WordPress/gutenberg/pull/36935))
- @jorgecontreras: Fix draft previews. ([37952](https://github.com/WordPress/gutenberg/pull/37952))
- @knilkantha: Scripts: Corrected "npm start:Hot" in the README file. ([38178](https://github.com/WordPress/gutenberg/pull/38178))
```

I envisage the release lead confidence checking all the PRs in the list (note it's generally fairly small) and then amending the list as required. I also envisage us creating a culture of ensuring PRs are labelled accordingly in order to give kudos to new contributors.

I acknowledge that this doesn't cover other types of contribution such as reviews...etc. However the concept could be extended as suggested in the discussion, namely by creating labels for `First-time reporter` and `First-time commenter` to track non-PR contributions as well (although that would require further tweaks to this script). I believe however that this PR is a good first step towards establishing a regular culture of recognising first time contributors. In short, it's not perfect, but it's **better than what we have right now**.

## Testing Instructions

1. Checkout PR.
2. Open terminal in Gutenberg repo.
3. Run `npm run changelog -- --milestone="Gutenberg 12.5"` (or use milestone of your choosing!).
4. See changelog generated with FTC section appended to the bottom.
5. Check the FTC's match up with [the labels in Github](https://github.com/WordPress/gutenberg/pulls?q=is%3Apr+milestone%3A%22Gutenberg+12.5%22+label%3A%22First-time+Contributor%22) - note I've filtered out "bots" from the list as they don't require any validation or recognition (yet 😉  ).

## Screenshots 

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
